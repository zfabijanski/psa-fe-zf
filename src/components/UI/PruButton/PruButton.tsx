import React from "react";
import styled, { CSSProp } from "styled-components/macro";
import { theme } from "../../../theme/theme";
import { Icon, TIconType, setIconColor } from "../Icon";

export type PruButtonType = keyof typeof config.color.default;

export interface IButtonProps {
  size?: "large" | "medium" | "small";
  buttonType?: PruButtonType;
  onClick?: React.MouseEventHandler<HTMLButtonElement>;
  disabled?: boolean;
  children?: React.ReactNode;
  htmlType?: "button" | "submit" | "reset";
  css?: CSSProp;
  icon?: TIconType | null;
  iconPosition?: "left" | "right";
}
const defaultProps: IButtonProps = {
  size: "large",
  buttonType: "primary",
};

export const PruButton: React.FC<Partial<IButtonProps>> = ({
  htmlType = "button",
  children,
  icon,
  iconPosition = "left",
  ...props
}) => (
  <StyledButton type={htmlType} {...defaultProps} {...props}>
    {!!icon && iconPosition !== "right" && (
      <StyledIcon position="left" name={icon} />
    )}
    {children}
    {!!icon && iconPosition === "right" && (
      <StyledIcon position="right" name={icon} />
    )}
  </StyledButton>
);

const StyledButton = styled.button<IButtonProps>`
  text-decoration: none;
  outline: 0px;
  cursor: pointer;
  box-sizing: border-box;

  display: flex;
  justify-content: center;
  align-items: center;

  min-width: 128px;
  max-width: 400px;
  height: ${({ size }) => config.height[size!]};
  padding-left: ${({ size }) => config.padding[size!]};
  padding-right: ${({ size }) => config.padding[size!]};

  font-size: ${({ size }) => config.fontSize[size!]};
  font-weight: 700;
  line-height: 20px;
  text-align: center;

  color: ${({ buttonType }) => config.color.default[buttonType!]};
  background-color: ${({ buttonType }) =>
    config.backgroundColor.default[buttonType!]};
  border: 2px solid
    ${({ buttonType }) => config.borderColor.default[buttonType!]};
  ${({ buttonType }) => setIconColor(config.color.default[buttonType!])};

  &:hover {
    color: ${({ buttonType }) => config.color.hover[buttonType!]};
    background-color: ${({ buttonType }) =>
      config.backgroundColor.hover[buttonType!]};
    border: 2px solid
      ${({ buttonType }) => config.borderColor.hover[buttonType!]};
    ${({ buttonType }) => setIconColor(config.color.hover[buttonType!])};
  }

  &:disabled {
    cursor: initial;
    color: ${({ buttonType }) => config.color.disabled[buttonType!]};
    background-color: ${({ buttonType }) =>
      config.backgroundColor.disabled[buttonType!]};
    border: 2px solid
      ${({ buttonType }) => config.borderColor.disabled[buttonType!]};
    ${({ buttonType }) => setIconColor(config.color.disabled[buttonType!])};
  }

  &:active:not(:disabled) {
    transform: translateY(1px);
  }

  ${(props) => props.css};
`;

const StyledIcon = styled(Icon)<{ position: "left" | "right" }>`
  margin-right: ${({ position }) => (position === "left" ? "12px" : "-4px")};
  margin-left: ${({ position }) => (position === "right" ? "12px" : "-4px")};
`;

const { newColors } = theme;

const config = {
  height: {
    large: "52px",
    medium: "44px",
    small: "28px",
  },
  padding: {
    large: "32px",
    medium: "32px",
    small: "24px",
  },
  fontSize: {
    large: "16px",
    medium: "16px",
    small: "14px",
  },
  color: {
    default: {
      primary: newColors.white100,
      primaryBlack: newColors.white100,
      secondary: newColors.grayDark,
      error: newColors.white100,
    },
    hover: {
      primary: newColors.white100,
      primaryBlack: newColors.white100,
      secondary: newColors.primary100,
      error: newColors.white100,
    },
    disabled: {
      primary: newColors.white100,
      primaryBlack: newColors.gray5,
      secondary: newColors.gray60,
      error: newColors.white100,
    },
  },
  backgroundColor: {
    default: {
      primaryBlack: newColors.grayDark,
      primary: newColors.primary100,
      secondary: newColors.white100,
      error: newColors.error,
    },
    hover: {
      primary: newColors.grayDark,
      primaryBlack: newColors.primary100,
      secondary: newColors.white100,
      error: newColors.error,
    },
    disabled: {
      primary: newColors.inactive,
      primaryBlack: newColors.gray60,
      secondary: newColors.gray10,
      error: newColors.error,
    },
  },
  borderColor: {
    default: {
      primary: newColors.primary100,
      primaryBlack: newColors.grayDark,
      secondary: newColors.gray30,
      error: newColors.error,
    },
    hover: {
      primary: newColors.grayDark,
      primaryBlack: newColors.primary100,
      secondary: newColors.primary100,
      error: newColors.error,
    },
    disabled: {
      primary: newColors.inactive,
      primaryBlack: newColors.gray60,
      secondary: newColors.gray30,
      error: newColors.error,
    },
  },
};

export default PruButton;
