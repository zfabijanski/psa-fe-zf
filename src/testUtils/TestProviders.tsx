import { render, RenderOptions } from "@testing-library/react";
import { FC, PropsWithChildren, ReactElement } from "react";
import { Provider } from "react-redux";
import { ThemeProvider } from "styled-components";
import { Normalize } from "styled-normalize";
import GlobalStyles from "theme/globalStyles";
import { theme } from "theme/theme";
import rootStore, { configureStore, rootReducer as reducer } from "AppStore";
import { BrowserRouter } from "react-router-dom";
import { Modal } from "components/UI/Modal";
import { IntlProvider } from "react-intl";
import { plMessages } from "translations/pl";
import I18nProvider from "I18nProvider";

interface TestProvidersProps {
  store?: typeof rootStore;
  renderModal?: boolean;
  realIntlProvider?: boolean;
}

export const addTestProviders =
  ({
    store = configureStore({ reducer }),
    renderModal = true,
    // @TODO: update tests to use realIntlProvider everywhere
    realIntlProvider = false,
  }: TestProvidersProps): FC<PropsWithChildren<{}>> =>
  ({ children }) => {
    const TranslationsProvider = realIntlProvider ? IntlProvider : I18nProvider;

    return (
      <Provider store={store}>
        <BrowserRouter>
          <ThemeProvider theme={theme}>
            <TranslationsProvider locale="pl" messages={plMessages}>
              <Normalize />
              <GlobalStyles />
              {children}
              {renderModal && <Modal />}
            </TranslationsProvider>
          </ThemeProvider>
        </BrowserRouter>
      </Provider>
    );
  };

const customRender = (
  ui: ReactElement,
  options: Omit<RenderOptions, "options"> = {},
  testProviderOptions: TestProvidersProps = {}
) => render(ui, { wrapper: addTestProviders(testProviderOptions), ...options });

export * from "@testing-library/react";

// override render method
export { customRender as render };
