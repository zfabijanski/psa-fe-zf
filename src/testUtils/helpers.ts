export const sleep = (ms: number = 100): Promise<unknown> =>
  new Promise((resolve) => setTimeout(resolve, ms));
