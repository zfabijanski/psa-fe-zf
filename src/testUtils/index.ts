export * from "./TestProviders";
export * from "./mockCreators";
export * from "./helpers";
