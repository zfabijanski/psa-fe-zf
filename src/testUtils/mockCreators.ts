import {
  ICalculatorCoversConfig,
  IProductCovers,
} from "features/Calculator/types";
import {
  DocumentState,
  IDirectory,
  IDirectoryMetadata,
  IDocumentMetadata,
} from "features/Library/types";
import { IMeeting } from "features/Meeting/History/types";
import { IIllustration, ProductGuid } from "features/Products/types";
import { IModalData, ModalTypes } from "models/Modal";
import {
  Frequency,
  IFundRiskProfileConfig,
  IProductConfig,
  IProductFund,
  IProductGuarantee,
  InvestmentProfile,
  Role,
} from "models/common";
import {
  CalculationStatus,
  Direction,
  ICalculationDetailsDto,
  ICover,
  IInsured,
  YesNo,
} from "models/calculator";

export const createCalculationDetailsDto = (
  overrides?: Partial<ICalculationDetailsDto>
): ICalculationDetailsDto => ({
  calculation_id: 1,
  working_date: "2020-01-01",
  premiums_total: 100,
  guaranteed_death_benefit: 100,
  guaranteed_maturity_benefit: 100,
  projected_maturity_value_egr_m: 100,
  product_categories: [],
  uuid: "uuid",
  version: "version",
  source_system: "source_system",
  product_guid: 1,
  insureds_list: [],
  covers_list: [],
  funds_list: [],
  ...overrides,
});

export const createInsured = (overrides?: Partial<IInsured>): IInsured => ({
  insured_idx: 1,
  insured_uuid: "insured_uuid",
  age_in_months: 1,
  date_of_birth: "2020-01-01",
  role: Role.MAIN,
  ...overrides,
});

export const createCover = (overrides?: Partial<ICover>): ICover => ({
  cover_id: 1,
  cover_idx: 1,
  insured_idx: 1,
  code: "code",
  frequency_code: Frequency.Annually,
  payments: 1,
  direction_of_calculation: Direction.SumAssuredToPremium,
  investment_profile: InvestmentProfile.Balanced,
  indexation: YesNo.Yes,
  sum_assured: 1,
  premium_per_freq: 1,
  duration: 1,
  idx_option: YesNo.Yes,
  rop_option: YesNo.Yes,
  add_sum_assured: 1,
  ending_age: 1,
  exp_mat_ben: 1,
  ...overrides,
});

export const createCalculatorCoversConfig = (
  overrides?: Partial<ICalculatorCoversConfig>
): ICalculatorCoversConfig => ({
  coverName: "coverName",
  isChild: false,
  isHospital: false,
  ...overrides,
});

export const createCalculatorCoversConfigMap = (
  overrides?: Partial<Record<string, ICalculatorCoversConfig>>
): Record<string, ICalculatorCoversConfig> => ({
  coverName1: createCalculatorCoversConfig({ coverName: "coverName1" }),
  coverName2: createCalculatorCoversConfig({ coverName: "coverName2" }),
  ...overrides,
});

export const createProductCovers = (
  overrides?: Partial<IProductCovers>
): IProductCovers => ({
  mainCover: "mainCover",
  additionalCovers: ["additionalCover1", "additionalCover2"],
  wopCovers: ["wopCover1", "wopCover2"],
  childCovers: ["childCover1", "childCover2"],
  ...overrides,
});

export const createProductCoversMap = (
  overrides?: Partial<Record<string, IProductCovers>>
): Record<string, IProductCovers> => ({
  mainCover1: createProductCovers({ mainCover: "mainCover1" }),
  mainCover2: createProductCovers({ mainCover: "mainCover2" }),
  ...overrides,
});

export const createProductConfig = (
  overrides?: Partial<IProductConfig>
): IProductConfig => ({
  product_guid: 1,
  product_name: "product_name",
  product_categories: [],
  product_freqs: [],
  product_covers: [],
  product_funds: [],
  ...overrides,
});

export const createProductConfigMap = (
  overrides?: Partial<Record<string, IProductConfig>>
): Record<string, IProductConfig> => ({
  1: createProductConfig({ product_name: "product_name1" }),
  2: createProductConfig({ product_name: "product_name2" }),
  ...overrides,
});

export const createFundRiskProfileConfig = (
  overrides?: Partial<IFundRiskProfileConfig>
): IFundRiskProfileConfig => ({
  code: "code",
  name_pl: "name_pl",
  name_en: "name_en",
  risk_level: 1,
  status_code: "status_code",
  notes: "notes",
  ...overrides,
});

export const createProductGuarantee = (
  overrides?: Partial<IProductGuarantee>
): IProductGuarantee => ({
  guarantee_code: "guarantee_code",
  guarantee_name: "guarantee_name",
  guarantee_min_value: "1",
  guarantee_duration_months: "4",
  guarantee_percentages: [{ guarantee_percent: 10 }],
  ...overrides,
});

export const createProductFund = (
  overrides?: Partial<IProductFund>
): IProductFund => ({
  cover_code: "cover_code",
  fund_code: "fund_code",
  fund_name: "fund_name",
  fund_risk_profile: "code",
  fund_order: 1,
  guarantees: [createProductGuarantee()],
  ...overrides,
});

export const createModalData = (
  overrides: Partial<IModalData> = {}
): IModalData => ({
  modalIllustration: "alert-fill",
  modalTitleTrKey: "modalTitleTrKey",
  modalContentTrKey: "modalContentTrKey",
  modalContent: "modalContent",
  modalButtons: {
    confirm: { textTrKey: "modal.ok" },
  },
  modalType: ModalTypes.default,
  maskClosable: true,
  ...overrides,
});

export const createIllustration = (
  overrides: Partial<IIllustration> = {}
): IIllustration => ({
  idd_id: 1,
  product_guid: ProductGuid.Savings,
  adequacy_id: 1,
  order_id: 1,
  expected_amount: { value: 3 },
  premium_total: { value: 3 },
  duration: { value: 3 },
  premium_main_cover: { value: 3 },
  frequency: { value: 3 },
  sum_of_premiums: { value: 3 },
  sum_assured: { value: 3 },
  investment_profile: { value: 3 },
  pru_calc_calculation_id: 1,
  status: CalculationStatus.Illustrated,
  json_needs_analysis: null,
  ...overrides,
});

export const createDirectory = (
  overrides?: Partial<IDirectory>
): IDirectory => ({
  directoryId: 1,
  directories: [],
  documentsIds: [1, 2, 3],
  ...overrides,
});

export const createDirectoryMetadata = (
  overrides?: Partial<IDirectoryMetadata>
): IDirectoryMetadata => ({
  directoryId: 1,
  name: "name",
  isOpen: false,
  ...overrides,
});

export const createDocumentMetadata = (
  overrides?: Partial<IDocumentMetadata>
): IDocumentMetadata => ({
  documentId: 1,
  guid: "guid",
  name: "name",
  state: DocumentState.NotAttached,
  isStateChanging: false,
  moduleName: "PRU_LIBRARY",
  ...overrides,
});

export const createMeeting = (overrides?: Partial<IMeeting>): IMeeting => ({
  meetingNo: "meetingNo",
  meetingId: 1,
  name: "name",
  age: 18,
  lastAccessDate: new Date("2023-01-01"),
  createDate: new Date("2023-01-01"),
  statusId: 1,
  statusName: "statusName",
  statusDescription: "statusDescription",
  state: "state",
  haveAdequacy: false,
  haveCalculation: false,
  haveIdd: false,
});
