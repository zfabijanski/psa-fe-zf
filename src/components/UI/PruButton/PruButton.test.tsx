import { PruButton, IButtonProps, PruButtonType } from "./PruButton";
import { render, screen, fireEvent } from "testUtils";

const mockOnClick = jest.fn();

const defaultProps: Partial<IButtonProps> = {
  onClick: mockOnClick,
};

const setup = (props: Partial<IButtonProps> = {}) =>
  render(<PruButton {...defaultProps} {...props} />);

describe("PruButton", () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it("should render correctly", () => {
    const { container } = setup();
    expect(container).toMatchSnapshot();
  });

  it("should call onClick when button is clicked", () => {
    setup();
    fireEvent.click(screen.getByRole("button"));
    expect(mockOnClick).toHaveBeenCalledTimes(1);
  });

  it("should handle different variants", () => {
    setup({ buttonType: "primary" });
    expect(screen.getByRole("button")).toHaveStyle(
      "background-color: rgb(5, 90, 96)"
    );
  });

  it("should render the text", () => {
    setup({ children: "text" });
    expect(screen.getByText("text")).toBeInTheDocument();
  });

  it.each(["primary", "primaryBlack", "secondary", "error"] as PruButtonType[])(
    `should render the %s button`,
    (buttonType: PruButtonType) => {
      setup({ buttonType });
      expect(screen.getByRole("button")).toMatchSnapshot();
    }
  );
});
