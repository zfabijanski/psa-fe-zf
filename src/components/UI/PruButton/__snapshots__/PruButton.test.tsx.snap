// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PruButton should render correctly 1`] = `
.c0 {
  -webkit-text-decoration: none;
  text-decoration: none;
  outline: 0px;
  cursor: pointer;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  min-width: 128px;
  max-width: 400px;
  height: 52px;
  padding-left: 32px;
  padding-right: 32px;
  font-size: 16px;
  font-weight: 700;
  line-height: 20px;
  text-align: center;
  color: #FFFFFF;
  background-color: #055A60;
  border: 2px solid #055A60;
}

.c0 svg g,
.c0 svg g path,
.c0 svg g circle,
.c0 svg g rect {
  stroke: #FFFFFF;
}

.c0 svg g rect {
  fill: #FFFFFF;
}

.c0 svg g tspan {
  stroke: transparent;
  fill: #FFFFFF;
}

.c0 svg > path {
  stroke: #FFFFFF;
}

.c0:hover {
  color: #FFFFFF;
  background-color: #181818;
  border: 2px solid #181818;
}

.c0:hover svg g,
.c0:hover svg g path,
.c0:hover svg g circle,
.c0:hover svg g rect {
  stroke: #FFFFFF;
}

.c0:hover svg g rect {
  fill: #FFFFFF;
}

.c0:hover svg g tspan {
  stroke: transparent;
  fill: #FFFFFF;
}

.c0:hover svg > path {
  stroke: #FFFFFF;
}

.c0:disabled {
  cursor: initial;
  color: #FFFFFF;
  background-color: #9BBDBF;
  border: 2px solid #9BBDBF;
}

.c0:disabled svg g,
.c0:disabled svg g path,
.c0:disabled svg g circle,
.c0:disabled svg g rect {
  stroke: #FFFFFF;
}

.c0:disabled svg g rect {
  fill: #FFFFFF;
}

.c0:disabled svg g tspan {
  stroke: transparent;
  fill: #FFFFFF;
}

.c0:disabled svg > path {
  stroke: #FFFFFF;
}

.c0:active:not(:disabled) {
  -webkit-transform: translateY(1px);
  -ms-transform: translateY(1px);
  transform: translateY(1px);
}

<div>
  <button
    class="c0"
    type="button"
  />
  <div />
</div>
`;

exports[`PruButton should render the error button 1`] = `
.c0 {
  -webkit-text-decoration: none;
  text-decoration: none;
  outline: 0px;
  cursor: pointer;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  min-width: 128px;
  max-width: 400px;
  height: 52px;
  padding-left: 32px;
  padding-right: 32px;
  font-size: 16px;
  font-weight: 700;
  line-height: 20px;
  text-align: center;
  color: #FFFFFF;
  background-color: #D13C15;
  border: 2px solid #D13C15;
}

.c0 svg g,
.c0 svg g path,
.c0 svg g circle,
.c0 svg g rect {
  stroke: #FFFFFF;
}

.c0 svg g rect {
  fill: #FFFFFF;
}

.c0 svg g tspan {
  stroke: transparent;
  fill: #FFFFFF;
}

.c0 svg > path {
  stroke: #FFFFFF;
}

.c0:hover {
  color: #FFFFFF;
  background-color: #D13C15;
  border: 2px solid #D13C15;
}

.c0:hover svg g,
.c0:hover svg g path,
.c0:hover svg g circle,
.c0:hover svg g rect {
  stroke: #FFFFFF;
}

.c0:hover svg g rect {
  fill: #FFFFFF;
}

.c0:hover svg g tspan {
  stroke: transparent;
  fill: #FFFFFF;
}

.c0:hover svg > path {
  stroke: #FFFFFF;
}

.c0:disabled {
  cursor: initial;
  color: #FFFFFF;
  background-color: #D13C15;
  border: 2px solid #D13C15;
}

.c0:disabled svg g,
.c0:disabled svg g path,
.c0:disabled svg g circle,
.c0:disabled svg g rect {
  stroke: #FFFFFF;
}

.c0:disabled svg g rect {
  fill: #FFFFFF;
}

.c0:disabled svg g tspan {
  stroke: transparent;
  fill: #FFFFFF;
}

.c0:disabled svg > path {
  stroke: #FFFFFF;
}

.c0:active:not(:disabled) {
  -webkit-transform: translateY(1px);
  -ms-transform: translateY(1px);
  transform: translateY(1px);
}

<button
  class="c0"
  type="button"
/>
`;

exports[`PruButton should render the primary button 1`] = `
.c0 {
  -webkit-text-decoration: none;
  text-decoration: none;
  outline: 0px;
  cursor: pointer;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  min-width: 128px;
  max-width: 400px;
  height: 52px;
  padding-left: 32px;
  padding-right: 32px;
  font-size: 16px;
  font-weight: 700;
  line-height: 20px;
  text-align: center;
  color: #FFFFFF;
  background-color: #055A60;
  border: 2px solid #055A60;
}

.c0 svg g,
.c0 svg g path,
.c0 svg g circle,
.c0 svg g rect {
  stroke: #FFFFFF;
}

.c0 svg g rect {
  fill: #FFFFFF;
}

.c0 svg g tspan {
  stroke: transparent;
  fill: #FFFFFF;
}

.c0 svg > path {
  stroke: #FFFFFF;
}

.c0:hover {
  color: #FFFFFF;
  background-color: #181818;
  border: 2px solid #181818;
}

.c0:hover svg g,
.c0:hover svg g path,
.c0:hover svg g circle,
.c0:hover svg g rect {
  stroke: #FFFFFF;
}

.c0:hover svg g rect {
  fill: #FFFFFF;
}

.c0:hover svg g tspan {
  stroke: transparent;
  fill: #FFFFFF;
}

.c0:hover svg > path {
  stroke: #FFFFFF;
}

.c0:disabled {
  cursor: initial;
  color: #FFFFFF;
  background-color: #9BBDBF;
  border: 2px solid #9BBDBF;
}

.c0:disabled svg g,
.c0:disabled svg g path,
.c0:disabled svg g circle,
.c0:disabled svg g rect {
  stroke: #FFFFFF;
}

.c0:disabled svg g rect {
  fill: #FFFFFF;
}

.c0:disabled svg g tspan {
  stroke: transparent;
  fill: #FFFFFF;
}

.c0:disabled svg > path {
  stroke: #FFFFFF;
}

.c0:active:not(:disabled) {
  -webkit-transform: translateY(1px);
  -ms-transform: translateY(1px);
  transform: translateY(1px);
}

<button
  class="c0"
  type="button"
/>
`;

exports[`PruButton should render the primaryBlack button 1`] = `
.c0 {
  -webkit-text-decoration: none;
  text-decoration: none;
  outline: 0px;
  cursor: pointer;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  min-width: 128px;
  max-width: 400px;
  height: 52px;
  padding-left: 32px;
  padding-right: 32px;
  font-size: 16px;
  font-weight: 700;
  line-height: 20px;
  text-align: center;
  color: #FFFFFF;
  background-color: #181818;
  border: 2px solid #181818;
}

.c0 svg g,
.c0 svg g path,
.c0 svg g circle,
.c0 svg g rect {
  stroke: #FFFFFF;
}

.c0 svg g rect {
  fill: #FFFFFF;
}

.c0 svg g tspan {
  stroke: transparent;
  fill: #FFFFFF;
}

.c0 svg > path {
  stroke: #FFFFFF;
}

.c0:hover {
  color: #FFFFFF;
  background-color: #055A60;
  border: 2px solid #055A60;
}

.c0:hover svg g,
.c0:hover svg g path,
.c0:hover svg g circle,
.c0:hover svg g rect {
  stroke: #FFFFFF;
}

.c0:hover svg g rect {
  fill: #FFFFFF;
}

.c0:hover svg g tspan {
  stroke: transparent;
  fill: #FFFFFF;
}

.c0:hover svg > path {
  stroke: #FFFFFF;
}

.c0:disabled {
  cursor: initial;
  color: #F8F8F8;
  background-color: #999999;
  border: 2px solid #999999;
}

.c0:disabled svg g,
.c0:disabled svg g path,
.c0:disabled svg g circle,
.c0:disabled svg g rect {
  stroke: #F8F8F8;
}

.c0:disabled svg g rect {
  fill: #F8F8F8;
}

.c0:disabled svg g tspan {
  stroke: transparent;
  fill: #F8F8F8;
}

.c0:disabled svg > path {
  stroke: #F8F8F8;
}

.c0:active:not(:disabled) {
  -webkit-transform: translateY(1px);
  -ms-transform: translateY(1px);
  transform: translateY(1px);
}

<button
  class="c0"
  type="button"
/>
`;

exports[`PruButton should render the secondary button 1`] = `
.c0 {
  -webkit-text-decoration: none;
  text-decoration: none;
  outline: 0px;
  cursor: pointer;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  min-width: 128px;
  max-width: 400px;
  height: 52px;
  padding-left: 32px;
  padding-right: 32px;
  font-size: 16px;
  font-weight: 700;
  line-height: 20px;
  text-align: center;
  color: #181818;
  background-color: #FFFFFF;
  border: 2px solid #E3E3E3;
}

.c0 svg g,
.c0 svg g path,
.c0 svg g circle,
.c0 svg g rect {
  stroke: #181818;
}

.c0 svg g rect {
  fill: #181818;
}

.c0 svg g tspan {
  stroke: transparent;
  fill: #181818;
}

.c0 svg > path {
  stroke: #181818;
}

.c0:hover {
  color: #055A60;
  background-color: #FFFFFF;
  border: 2px solid #055A60;
}

.c0:hover svg g,
.c0:hover svg g path,
.c0:hover svg g circle,
.c0:hover svg g rect {
  stroke: #055A60;
}

.c0:hover svg g rect {
  fill: #055A60;
}

.c0:hover svg g tspan {
  stroke: transparent;
  fill: #055A60;
}

.c0:hover svg > path {
  stroke: #055A60;
}

.c0:disabled {
  cursor: initial;
  color: #999999;
  background-color: #F3F3F3;
  border: 2px solid #E3E3E3;
}

.c0:disabled svg g,
.c0:disabled svg g path,
.c0:disabled svg g circle,
.c0:disabled svg g rect {
  stroke: #999999;
}

.c0:disabled svg g rect {
  fill: #999999;
}

.c0:disabled svg g tspan {
  stroke: transparent;
  fill: #999999;
}

.c0:disabled svg > path {
  stroke: #999999;
}

.c0:active:not(:disabled) {
  -webkit-transform: translateY(1px);
  -ms-transform: translateY(1px);
  transform: translateY(1px);
}

<button
  class="c0"
  type="button"
/>
`;
